### ISSUE TITLE
cbindgen emits constants in spite of no-export and/or ignore annotations 

### ISSUE THREAD

**Issue by scovich** [mozilla/cbindgen#947]
For the following rust code:
```rust
// cbindgen:ignore
// cbindgen:no-export
pub struct Foo;

// cbindgen:ignore
// cbindgen:no-export
impl Foo {
    // cbindgen:ignore
    // cbindgen:no-export
    pub const BAR: i64 = 42;
}
```
cbindgen still emits the following header code:
```c++
constexpr static const int64_t Foo_BAR = 42;
```
I haven't found any combination of annotations that can suppress it, even tho the docs suggest it should work?

For [ignore](https://github.com/mozilla/cbindgen/blob/master/docs.md#ignore-annotation):
> You can manually ignore other stuff with the `ignore` annotation attribute

For [no-export](https://github.com/mozilla/cbindgen/blob/master/docs.md#no-export-annotation):
> cbindgen will usually emit all items it finds, as instructed by the parse and export config sections. This annotation will make cbindgen skip this item from the output, while still being aware of it.

**Comment by emilio** [comment]
I think you need to use triple-dash, otherwise the comment isn't preserved by syn

**Comment by emilio** [comment]
Triple slash that is, sorry.

**Comment by scovich** [comment]
This still emits:
```rust
pub struct Foo;

impl Foo {
    /// cbindgen:ignore
    /// cbindgen:no-export
    pub const BAR: i64 = 42;
}
```
This suppresses it:
```rust
pub struct Foo;

/// cbindgen:ignore
/// cbindgen:no-export
impl Foo {
    pub const BAR: i64 = 42;
}
```
So it still seems like the constant isn't paying attention to the directive -- rather the struct impl is?

**Comment by scovich** [comment]
Starting exploring in https://github.com/mozilla/cbindgen/pull/949

It turns out `/// cbindgen:XXX` annotations don't affect anything inside a struct impl block -- in fact, they're not even parsed (verified by inserting a `panic!` at the `NameValue` "doc" case of [is_skip_item_attr](https://github.com/mozilla/cbindgen/blob/ca78140c01518a655355f84da1f3872939123b66/src/bindgen/utilities.rs#L100))

**Comment by scovich** [comment]
I was able to find and add the missing `should_skip_parsing` calls, which fixes the `cbindgen:ignore` case. Still looking for the code that processes `cbindgen:no-export`, tho.

**Comment by emilio** [comment]
Fixed by #949.
